vhdl_syntax/src/parser/productions/compositve_types.rs

```diff
@@ -0,0 +1,252 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c)  2025, Lukas Scheller lukasscheller@icloud.com
+/// Parsing of composite types (LRM §5.3)
+use crate::parser::Parser;
+use crate::syntax::node_kind::NodeKind::*;
+use crate::tokens::token_kind::Keyword as Kw;
+use crate::tokens::TokenKind::*;
+use crate::tokens::TokenStream;
+
+impl<T: TokenStream> Parser<T> {
+    pub fn array_type_definition(&mut self) {
+        self.start_node(ArrayTypeDefinition);
```

Comment:
In your implementation, you tried separating between unbounded and constrained array definitions. Woluld you prefer that?

Reply:
Yes. The main motivation was to be as compatible to the standard as possible.